syft/rekor/queryRekor/utils.go

```diff
@@ -0,0 +1,274 @@
+		// this decodes the base64 string
+		attDecoded := string(attAnon.Data)
+
+		att := new(inTotoAttestation)
+		err := json.Unmarshal([]byte(attDecoded), att)
+		if err != nil {
+			log.Debug("Error parsing attestation")
+			return nil, err
+		}
+
+		// validation the entry is an sbom entry is done by validating attestation struct
+		if att.PredicateType == "" {
+			log.Debugf("Rekor entry %v could not be parsed as an sbom entry", *entry.LogIndex)
+			continue
+		}
+		if att.PredicateType != "http://lumjjb/sbom-documents" {
+			log.Debugf("Rekor entry %v could not be parsed as an sbom entry", *entry.LogIndex)
+			continue
+		}
+		if att.Predicate.Sboms == nil {
+			log.Debugf("Rekor entry %v could not be parsed as an sbom entry", *entry.LogIndex)
+			continue
+		}
+
+		atts = append(atts, *att)
+	}
+
+	return atts, nil
+}
+
+func parseSboms(spdxBytesList map[*inTotoAttestation][]byte) (map[*inTotoAttestation]spdx.Document2_2, error) {
+
+	sboms := make(map[*inTotoAttestation]spdx.Document2_2)
+	for att, spdxBytes := range spdxBytesList {
+		//remove all SHA512 hashes because spdx/tools-golang does not support
+		// PR fix is filed but not merged
+
+		regex, err := regexp.Compile("\n.*SHA512.*")
+		if err != nil {
+			log.Debug("Error decoding sbom to syft type")
+			return nil, err
+		}
+
+		spdxBytes = regex.ReplaceAll(spdxBytes, nil)
+		sbom, err := tvloader.Load2_2(bytes.NewReader(spdxBytes))
+		if err != nil {
+			log.Debug("Error loading sbom bytes into spdx type")
+			return nil, err
+		}
+
+		sboms[att] = *sbom
+	}
+
+	return sboms, nil
+}
+
+func formatSbomPackageName(att *inTotoAttestation) string {
+	repo := att.Predicate.BuildMetadata.ArtifactSourceRepo
+	commit := att.Predicate.BuildMetadata.ArtifactSourceRepoCommit
+	if repo == "" {
+		return ""
+	}
+	if commit == "" {
+		return repo
+	}
+	return fmt.Sprintf("%v@%v", repo, commit)
+}
+
+func joinSbomsWithMetadata(sboms map[*inTotoAttestation]spdx.Document2_2) []spdxSbomWithMetadata {
+
+	var sbomsWithMetadata []spdxSbomWithMetadata
+	for att, sbom := range sboms {
+
+		packageName := formatSbomPackageName(att)
+		namespace := sbom.CreationInfo.DocumentNamespace
+
+		if packageName == "" {
+			packageName = namespace
+		}
+		if namespace == "" {
+			log.Debugf("sbom from rekor entry is malformed (does not have a namespace)")
+		} else {
+			sbomsWithMetadata = append(sbomsWithMetadata, spdxSbomWithMetadata{sbom: sbom, namespace: namespace, packageName: packageName})
+		}
+	}
+
+	return sbomsWithMetadata
+}
+
+// map a list of digests to a (sha1, sha256) tuple
+func parseDigests(digests []file.Digest) (string, string) {
+	var sha1 string
+	var sha256 string
+	if digests[0].Algorithm == "sha1" && digests[1].Algorithm == "sha256" {
+		sha1 = digests[0].Value
+		sha256 = digests[1].Value
+	} else if digests[0].Algorithm == "sha256" && digests[1].Algorithm == "sha1" {
+		sha256 = digests[0].Value
+		sha1 = digests[1].Value
+	} else {
+		log.Debug("Unexpected digests")
+		return "", ""
+	}
```

Comment:
hmm.. I think it may be cleaner to do a helper function or a closure here and then run the function on each digest. Probably want to go through all digests as well, as there would be more algs used in the future.